packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -0,0 +1,244 @@
+   */
+  rowHeaderIndex: PropTypes.number,
+};
+
+const defaultProps = {
+  rowHeaderIndex: 0,
+  columnHeaderDefaultWidth: '200px',
+  columnHeaderHeight: '2.5rem',
+  rowDefaultHeight: '2.5rem',
+};
+
+class WorklistDataGrid extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.focussedRow = 0;
+    this.focussedCol = 0;
+  }
+
+  componentDidMount() {
+    this.grid = document.querySelector(`#${this.props.id}`);
+
+    const focussedCell = this.grid.rows[this.focussedRow].cells[this.focussedCol];
+    focussedCell.setAttribute('tabIndex', '0');
+
+    // add event listeners to handle user key events in the grid.
+    this.grid.addEventListener('keydown', this.handleKeyDown);
+    this.grid.addEventListener('click', this.handleClick);
+  }
+
+  handleKeyDown(event) {
+    if (event.target?.parentNode?.parentNode?.parentNode?.id !== this.props.id) {
+      return;
+    }
+
+    const currentFocussedRow = event.target.parentElement.rowIndex;
+    const currentFocussedCol = event.target.cellIndex;
+
+    let nextRow = currentFocussedRow;
+    let nextCol = currentFocussedCol;
+
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_UP:
+        nextRow -= 1;
+        break;
+      case KeyCode.KEY_DOWN:
+        nextRow += 1;
+        break;
+      case KeyCode.KEY_LEFT:
+        nextCol -= 1;
+        break;
+      case KeyCode.KEY_RIGHT:
+        nextCol += 1;
+        break;
+      case KeyCode.KEY_HOME:
+        nextCol = 0;
+        if (event.ctrlKey) {
+          nextRow = this.getNumberOfHeaderRows();
+        }
+        break;
+      case KeyCode.KEY_END:
+        nextCol = this.props.rows[0].cells.length - 1;
+        if (event.ctrlKey) {
+          nextRow = (this.props.rows.length - 1) + this.getNumberOfHeaderRows();
+        }
+        break;
+      default:
+        return;
+    }
+    if (nextRow >= this.grid.rows.length || nextCol >= this.grid.rows[0].cells.length) {
+      event.preventDefault();
+      return;
+    }
+    if (nextCol < 0 || nextRow < 0) {
+      event.preventDefault();
+      return;
+    }
+
+    // Remove the ability to receive visible focus as only one cell should be able to receive tab focus.
+    // This is also how tabbing back to the grid ensures the grid last cell with the focus will be re-entered.
+    this.grid.rows[currentFocussedRow].cells[currentFocussedCol].setAttribute('tabIndex', '-1');
+
+    this.focussedRow = nextRow;
+    this.focussedCol = nextCol;
+
+    const elementWithTabStop = this.grid.rows[nextRow].cells[nextCol];
+    elementWithTabStop.setAttribute('tabIndex', '0');
+    elementWithTabStop.focus();
+    event.preventDefault();
+  }
+
+  handleClick(event) {
+    // Is this event for something that is on the grid
+    if (event.target?.parentNode?.parentNode?.parentNode?.id !== this.props.id) {
```

Comment:
This logic makes me think that the event listener is at the wrong level.

Reply:
Please see response in the question below.